Archipelago.MultiClient.Net.Tests/DataPackageFileSystemCacheFixture.cs

```diff
@@ -0,0 +1,242 @@
+                DataPackage = new DataPackage {
+                    Version = 5,
+                    Games = new Dictionary<string, GameData> {
+                        { "One", new TestGameData(2) }
+                    }
+                }
+            };
+
+            var socket = Substitute.For<IArchipelagoSocketHelper>();
+            var fileSystemDataPackageProvider = Substitute.For<IFileSystemDataPackageProvider>();
+            fileSystemDataPackageProvider.TryGetDataPackage(out _)
+                .Returns(x => {
+                    x[0] = localDataPackage;
+                    return true;
+                });
+
+            var sut = new DataPackageFileSystemCache(socket, fileSystemDataPackageProvider);
+
+            socket.PacketReceived += Raise.Event<ArchipelagoSocketHelper.PacketReceivedHandler>(serverDataPackage);
+
+            fileSystemDataPackageProvider.Received().SaveDataPackageToFile(Arg.Is<DataPackage>(p =>
+                p.Version == 5
+                && p.Games.Count == 2
+                && p.Games["One"].Version == 2
+                && p.Games["Two"].Version == 3
+            ));
+
+            sut.TryGetDataPackageFromCache(out var inMemoryDataPackage);
+
+            Assert.IsTrue(inMemoryDataPackage.Version == 5
+                          && inMemoryDataPackage.Games.Count == 2
+                          && inMemoryDataPackage.Games["One"].Version == 2
+                          && inMemoryDataPackage.Games["Two"].Version == 3);
+        }
+
+        [Test]
+        public void Should_not_save_game_when_version_is_0_but_still_keep_it_in_memory()
+        {
+            var localDataPackage = new DataPackage
+            {
+                Version = 2,
+                Games = new Dictionary<string, GameData> {
+                    { "One", new TestGameData(2) }
+                }
+            };
+
+            var serverDataPackage = new DataPackagePacket
+            {
+                DataPackage = new DataPackage
+                {
+                    Version = 0,
+                    Games = new Dictionary<string, GameData> {
+                        { "One", new TestGameData(3) },
+                        { "Two", new TestGameData(0) }
+                    }
+                }
+            };
+
+            var socket = Substitute.For<IArchipelagoSocketHelper>();
+            var fileSystemDataPackageProvider = Substitute.For<IFileSystemDataPackageProvider>();
+            fileSystemDataPackageProvider.TryGetDataPackage(out _)
+                .Returns(x => {
+                    x[0] = localDataPackage;
+                    return true;
+                });
+
+            var sut = new DataPackageFileSystemCache(socket, fileSystemDataPackageProvider);
+
+            socket.PacketReceived += Raise.Event<ArchipelagoSocketHelper.PacketReceivedHandler>(serverDataPackage);
+
+            fileSystemDataPackageProvider.Received().SaveDataPackageToFile(Arg.Is<DataPackage>(p =>

```

Comment:
This doesn't seem to assert that "Two" is NOT saved to file system. Did I miss something?

Reply:
p.Games.Count == 1, so it assets that there is only one game in the Games dictionary